### PR TITLE
XWIKI-12565: Replace "Document" with "Page" in the Index Application

### PR DESCRIPTION
**Pull request by gabrielasmeria**

### CODE REVIEW

**Comment by vmassol** [comment]
Gaby, could you also verify that this doesn't break any func test? Thanks!

**Comment by gabrielasmeria** [comment]
So far, I didn't find any of those strings in the tests. 
I'll be back ASAP with another comment after I'll run the tests, since I had some problems with Eclipse.

**Comment by gabrielasmeria** [comment]
All the tests passed. Thanks!

File: xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-ui/src/main/resources/XWiki/DocumentTreeTranslations.xml [hunk 1]
```diff
@@ -72,21 +72,21 @@ rendering.macro.documentTree.parameter.checkboxes.description=Whether to display
...
+rendering.macro.documentTree.parameter.hierarchyMode.name=Page Hierarchy Mode
+rendering.macro.documentTree.parameter.hierarchyMode.description=Specifies which hierarchy to use between pages. Possible values are "reference" (default) and "parentchild". If "reference" is used then the page hierarchy is defined using only the page reference, especially the nested spaces component of the page reference. If "parentchild" is used then the hierarchy is based on the parent/child relationship defined by the "parent" page field from the database.
 
 index.documentTree.empty=No pages found
 
 ## Deprecated translation keys
 
 ## Until 7.2M2
-rendering.macro.documentTree.parameter.showChildDocuments.name=Show Child Documents
```
Comment (evalica) [inline xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-ui/src/main/resources/XWiki/DocumentTreeTranslations.xml]: We don't modify the deprecated keys

**Comment by evalica** [comment]
Another change we need to make is 'Document Index' to 'Page Index'

**Comment by gabrielasmeria** [comment]
Fixed both problems. Thanks for noticing!

**Pull request merged**